presto-spark/src/main/java/com/facebook/presto/spark/planner/SparkRddPlanner.java

```diff
@@ -0,0 +1,254 @@
+                        .flatMap(taskSource -> taskSource.getSplits().stream()
+                                .map(split -> createTaskDescriptor(
+                                        fragment,
+                                        ImmutableList.of(
+                                                new TaskSource(
+                                                        taskSource.getPlanNodeId(),
+                                                        ImmutableSet.of(split),
+                                                        taskSource.getNoMoreSplitsForLifespan(),
+                                                        taskSource.isNoMoreSplits())))))
+                        .collect(toImmutableList());
+                List<byte[]> serializedRequests = taskRequests.stream()
+                        .map(sparkTaskRequestJsonCodec::toJsonBytes)
+                        .collect(toImmutableList());
+                return sparkContext.parallelize(serializedRequests)
+                        .flatMapToPair(createTaskProcessor(compilerFactory, taskStatsCollector));
+            }
+
+            List<SubPlanWithTaskSources> children = subPlan.getChildren();
+            checkArgument(
+                    remoteSources.size() == children.size(),
+                    "number of remote sources doesn't match the number of child stages: %s != %s",
+                    remoteSources.size(),
+                    children.size());
+
+            if (children.size() == 1) {
+                // Single remote source
+                SubPlanWithTaskSources childSubPlan = getOnlyElement(children);
+                JavaPairRDD<Integer, byte[]> childRdd = createRdd(childSubPlan);
+
+                PlanFragment childFragment = childSubPlan.getFragment();
+                RemoteSourceNode remoteSource = getOnlyElement(remoteSources);
+                List<PlanFragmentId> sourceFragmentIds = remoteSource.getSourceFragmentIds();
+                checkArgument(sourceFragmentIds.size() == 1, "expected to have exactly only a single source fragment");
+                checkArgument(childFragment.getId().equals(getOnlyElement(sourceFragmentIds)));
+
+                SparkTaskDescriptor sparkTaskDescriptor = createTaskDescriptor(fragment, ImmutableList.of());
+                byte[] serializedTaskDescriptor = sparkTaskRequestJsonCodec.toJsonBytes(sparkTaskDescriptor);
+
+                PartitioningHandle partitioning = fragment.getPartitioning();
+                if (partitioning.equals(COORDINATOR_DISTRIBUTION)) {
+                    // TODO: We assume COORDINATOR_DISTRIBUTION always means OutputNode
+                    // But it could also be TableFinishNode, in that case we should do collect and table commit on coordinator.
+
+                    // TODO: Do we want to return an RDD for root stage? -- or we should consider the result will not be large?
+                    List<Tuple2<Integer, byte[]>> collect = childRdd.collect();
```

Comment:
>> // TODO: Do we want to return an RDD for root stage? -- or we should consider the result will not be large?

Since this is planner code, I agree that this should not execute the RDD. 

In general, I am concerned about the result size. Usually the driver does not have much memory to hold a lot of data and we would need to have a LIMIT applied to the results to solve that. Would that be done internally by Presto already or we will have to put that in presto-spark module ?

Reply:
I assume that you want some Iterator at L180. Maybe create a lazy row iterator which will do rdd.collect() when we start consuming data off the iterator

Reply:
I applied this change: https://github.com/prestodb/presto/pull/13895/commits/5492e1703359ca34c8926051c280278475531c0e. Thanks @wenleix !

Reply:
@tejasapatil: 
> In general, I am concerned about the result size. Usually the driver does not have much memory to hold a lot of data and we would need to have a LIMIT applied to the results to solve that. 

The data getting collected are the written file information (think about the path/partition information) so Presto can make the Metastore call . I assume it's reasonable to collect such information on Spark Driver?